java/managed-vms/README.md

```diff
@@ -107,6 +118,47 @@ This app is a [Jetty](http://www.eclipse.org/jetty/) based [Servlet](http://www.
 
 1. go to the new default module which will be displayed in results from the deploy.  It will look like: `https://20150624t111224-dot-default-dot-PROJECTID.appspot.com`  -- This is not the default instance, and if you deploy you will need to manage your instances to both set default and delete unneeded instances.
 
+## Deploying the AppEngine Runtime
+
+1. Build the Docker Image for this project
+
+ `cd gae-docker; docker build -t gae-4bt .;cd ../bigtable-hello`
+
+1. Edit `src/main/java/com.example.cloud.bigtable.helloworld/BigtableHelper.java` to set `PROJECT_ID`, `CLUSTER_UNIQUE_ID`, and `ZONE` (if necessary) 
+
+1. Edit `src/main/webapp/index.html` to set `google-signin-client_id` 
+
+1. Edit `Dockerfile` and change the **`FROM`** to be the just built `gae-4bt` image.
+
+1. Build the java artifacts and docker image
+ 
+    `mvn clean compile process-resources war:exploded && docker build -t bigtable-hello .`<br />
+
+1. Deploy the application
+
+ `gcloud preview app deploy app.yaml`
+
+## AppEngine Debugging Hints
+The first thing to do, if you'd like to debug is use the `servlet.log()` methods, they seem to work when other loggers don't.  Then take control of your GAE instance:
+
+1. Find your instance
+  `gcloud preview app modules list`
+
+1. [Change the management of the instances](https://cloud.google.com/appengine/docs/managed-vms/access#changing_management)
+
+1. [SSH to the instance](https://cloud.google.com/sdk/gcloud/reference/compute/ssh)
+
+1. [Find the Container](https://cloud.google.com/appengine/docs/managed-vms/access#accessing_the_docker_container_in_production)
+
+1. Either show the container log  `docker logs <containerID>` or enter the container `docker exec -it <containerID> /bin/bash
+
+
+## Using Bigtable-Hello
+
+1. With your browser, go to [docker:8080](docker:8080) (Mac) or [localhost:8080](localhost:8080) (Linux) in your browser. (Local)  Or to https://<projectID>.appspot.com
+
+1. Sign-in with Google  It should could your visit. 
+
 ## Using JSON
```

Comment:
It's not clear how this part relates to the "count your visit" part of the app. Perhaps the beginning of the README could clarify that this app really provides two things: 1) A REST interface that can read and write arbitrary data to a Cloud Bigtable table. 2) A web interface that uses Cloud Bigtable to track the number of visits from your Google account.

Also, since this app opens up a couple of tables to the Internet at large, it might be good to add a brief security warning near the beginning of the README.